sky/packages/updater/lib/bundle.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:yaml/yaml.dart' as yaml;
+
+const String kBundleMagic = '#!mojo mojo:sky_viewer\n';
+
+Future<List<int>> _readBytesWithLength(RandomAccessFile file) async {
+  var buffer = new ByteData(4);
+  await file.readInto(buffer.buffer.asUint8List());
+  int length = buffer.getUint32(0, Endianness.LITTLE_ENDIAN);
+  return await file.read(length);
+}
+
+// Represents a parsed .flx Bundle. Contains information from the bundle's
+// header, as well as an open File handle positioned where the zip content
+// begins.
+// The bundle format is:
+// #!mojo mojo:sky_viewer\n
+// <32-bit length><signature of the manifest data>
+// <32-bit length><manifest data>
```

Comment:
In other places we're currently using YAML we're hitting issues where we want to programatically update the contents and are unable since the Dart library can parse a YAML file but can't write out one, in general.